docs/02-iaas/security-groups.md

```diff
@@ -0,0 +1,147 @@
+# Best Practise: How to configure and use security groups
+
+Security groups in OpenStack are part of the network security mechanisms provided for the users.
+They resemble sets of simple firewall rules allowing specific network traffic at a Port of a VM that connects it to a network.
+The rules allow specific network port numbers and protocols while also differentiating between ingress and egress directions.
+Usually security groups are assigned to the Port(s) when a VM is created, but assignments can also be changed at runtime later on.
+Multiple security groups can be assigned to a VM or Port simultaneously and in such case they will be combined as the union of all their rules.
+
+:::caution
+
+Security groups are mutable resources.
+Their rules can be adjusted at any time after creation.
+**Changing the rules of a security group will immediately apply the changes to all Ports or VMs it is assigned to.**
+It is advisable to always review resources which use a security group before making changes to it.
+
+:::
+
+## Gather Requirements of Your Setup
+
+Every virtual machine that is created may need different firewall rules.
+These requirements can also change over time.
+Adding or removing Security Groups will allow users to adjust the firewall rules specifically to their virtual machines.
+
+To harden the firewall settings for your virtual machine you may follow these steps:
+
+1. Before creating a virtual machine their purpose is already known. Identify all incoming and outgoing traffic that is needed.
+2. Look through already existing Security Groups and their rules. If a Security Group allows more traffic than needed it SHOULD NOT be used. If it contains only a few of the required rules it can be used in combination with other Security Groups, that have rules that fulfill the rest of the required traffic rules from point 1.
+3. If you were not successful in finding fitting security Groups or you need an additional special rule, you should create one or more new Security Groups in which you can set the reuquired rules.
+4. After having one or more Security Groups that fulfill your requirements, you can create the VM with those Security Groups already specified.
+
+### Special Cases
+
+When talking about network security it does not only rely on firewall rules but also on other configurations or time-crucial allowances.
+
+**SSH**
+SSH is needed on many virtual machines to operate its workload.
```

Comment:
```suggestion
SSH is needed on many virtual machines to operate their guest operating system.
```

Workloads (as in deployed software and services) often have web UIs and frontends outside of SSH. I think SSH is only relevant for the base system.